artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -174,6 +184,8 @@ public class ServerSessionPacketHandler implements ChannelHandler {
 
    private final Pool<NullResponseMessage_V2> poolNullResponseV2;
 
+   private final Map<Integer, String> producers = new HashMap<>();
```

Comment:
can you instead use io.netty.util.collection.IntObjectHashMap please? It's a lot better for this case here. It's a simple replacement and it's just like a regular HashMap.